### ISSUE TITLE
bug: BuiltinModuleError in PyTorch 2.3.0

### ISSUE THREAD

**Issue by sbrugman** [mkdocstrings/griffe#323]
Amazing work on `griffe`!

### Description of the bug

For `torch` 2.3.0 we get the following error:

```text
  File "/Users/me/Documents/code/project/venv/lib/python3.10/site-packages/_griffe/models.py", line 1503, in filepath
    raise BuiltinModuleError(self.name)
_griffe.exceptions.BuiltinModuleError: _profiler
```

Version 2.4.0 runs without issue.

### To Reproduce

 ```
 pip install "torch==2.3.0"
 griffe dump torch
 ```

### Full traceback

<details><summary>Full traceback</summary>

```python
INFO       Loading package torch
Traceback (most recent call last):
  File "/Users/me/Documents/code/project/venv/bin/griffe", line 8, in <module>
    sys.exit(main())
  File "/Users/me/Documents/code/project/venv/lib/python3.10/site-packages/_griffe/cli.py", line 560, in main
    return commands[subcommand](**opts_dict)
  File "/Users/me/Documents/code/project/venv/lib/python3.10/site-packages/_griffe/cli.py", line 387, in dump
    loader = _load_packages(
  File "/Users/me/Documents/code/project/venv/lib/python3.10/site-packages/_griffe/cli.py", line 99, in _load_packages
    loader.load(package, try_relative_path=True, find_stubs_package=find_stubs_package)
  File "/Users/me/Documents/code/project/venv/lib/python3.10/site-packages/_griffe/loader.py", line 179, in load
    top_module = self._load_package(package, submodules=submodules)
  File "/Users/me/Documents/code/project/venv/lib/python3.10/site-packages/_griffe/loader.py", line 511, in _load_package
    top_module = self._load_module(package.name, package.path, submodules=submodules)
  File "/Users/me/Documents/code/project/venv/lib/python3.10/site-packages/_griffe/loader.py", line 535, in _load_module
    return self._load_module_path(module_name, module_path, submodules=submodules, parent=parent)
  File "/Users/me/Documents/code/project/venv/lib/python3.10/site-packages/_griffe/loader.py", line 565, in _load_module_path
    self._load_submodules(module)
  File "/Users/me/Documents/code/project/venv/lib/python3.10/site-packages/_griffe/loader.py", line 570, in _load_submodules
    self._load_submodule(module, subparts, subpath)
  File "/Users/me/Documents/code/project/venv/lib/python3.10/site-packages/_griffe/loader.py", line 621, in _load_submodule
    parent_module.set_member(submodule_name, submodule)
  File "/Users/me/Documents/code/project/venv/lib/python3.10/site-packages/_griffe/mixins.py", line 208, in set_member
    if value.is_module and value.filepath != member.filepath:
  File "/Users/me/Documents/code/project/venv/lib/python3.10/site-packages/_griffe/models.py", line 1503, in filepath
    raise BuiltinModuleError(self.name)
_griffe.exceptions.BuiltinModuleError: _profiler
```

</details>

### Expected behavior

Output the json information, e.g.

```json
{
  "torch": {
    "docstring": {
      "endlineno": 11,
      "lineno": 3,
      "value": "The torch package contains data structures for multi-dimensional\ntensors and defines mathematical operations over these tensors.\nAdditionally, it provides many utilities for efficient serialization of\nTensors and arbitrary types, and other useful utilities.\n\nIt has a CUDA counterpart, that enables you to run your tensor computations\non an NVIDIA GPU with compute capability >= 3.0."
    },
    "filepath": "/Users/me/Documents/code/project/venv/lib/python3.10/site-packages/torch/__init__.py",
    "kind": "module",
    "labels": [],
 ...
```

### Environment information

- __System__: macOS-14.6.1-arm64-arm-64bit
- __Python__: cpython 3.10.13 (/Users/me/Documents/code/project/venv/bin/python)
- __Environment variables__:
- __Installed packages__:
  - `griffe` v1.2.0

**Comment by pawamoy** [comment]
Hey, thanks for the kind words and for the report!

It looks like the traceback is incomplete? I'll try to reproduce it anyway, the given info should be enough :slightly_smiling_face:

**Comment by sbrugman** [comment]
Hey, the three lines posted above were missing. Added them now. Again, great tool!

**Comment by pawamoy** [comment]
So, with only torch==2.3.0 (no dependencies), Griffe loads and dumps it without any issue. If I install torch's dependencies as well, I'm getting an error similar to yours: `_griffe.exceptions.BuiltinModuleError: _cudnn`. I suppose that if I fix mine, yours should be fixed too. Continuing investigation.

**Comment by pawamoy** [comment]
Should be fixed in v1.3.0 :slightly_smiling_face:
